### PR TITLE
Add tests for edge lookup in `GenericGraph.__getitem__()`

### LINKED ISSUE #3798: Add tuple key support in `GenericGraph.__getitem__()`
## Description of proposed feature

```py
vertices = [1, 2, 3, 4]
edges = [(1, 2), (2, 3), (3, 4), (1, 3), (1, 4)]
g = Graph(vertices, edges)
assert isinstance(g[1], Dot)  # This is OK
assert isinstance(g[(1, 2)], Line)  # But this will raise a KeyError
```

This is confusing. Why not add a support for edge mapping?

## How can the new feature be used?
Every time people want to get or set a edge, just write `g[(1, 2)]` instead of `g.edges[(1, 2)]`.

### PR DESCRIPTION
**Pull request by sridhar-3009**
_EDIT by chopan050: an older PR #3799 was merged which allowed indexing edges of a graph, but the tests in this PR are useful._

## Summary

Fixes #3798.

`GenericGraph.__getitem__` only looked up vertices. When a user passed a
`(u, v)` tuple to retrieve an edge mobject, it raised a `KeyError` against
`self.vertices` instead of looking up `self.edges`.

```python
g = Graph([1, 2, 3, 4], [(1, 2), (2, 3)])
g[1]       # works — returns vertex mobject
g[(1, 2)]  # before: KeyError; after: returns edge mobject
```

## Changes

`manim/mobject/graph.py`:
- Route 2-tuple keys to `self.edges` in `__getitem__`; all other keys
  continue to look up `self.vertices`

`tests/module/mobject/test_graph.py`:
- Add `test_graph_getitem` covering vertex lookup, edge lookup on
  `Graph`, and edge lookup on `DiGraph`

## Test plan

- [ ] `pytest tests/module/mobject/test_graph.py::test_graph_getitem`

### CODE REVIEW

File: manim/mobject/graph.py [hunk 1]
```diff
@@ -670,6 +670,8 @@ def _populate_edge_dict(
         raise NotImplementedError("To be implemented in concrete subclasses")
 
     def __getitem__(self: Graph, v: Hashable) -> Mobject:
+        if isinstance(v, tuple) and len(v) == 2:
+            return self.edges[v]
         return self.vertices[v]
```
Comment (chopan050) [inline manim/mobject/graph.py]: One thing I mentioned in issue #3798 is that vertices can be any hashable object. This includes tuples: vertices can be tuples themselves. This is not too strange, actually. For example, vertices could represent 2D coordinates. There's also this concept of a "line graph L of a graph G" where each vertex of L is an edge of G: if there is an edge connecting vertices 1 and 3 of G, then L has a vertex (1, 3), and so on. [`networkx` actually implements this.](https://networkx.org/documentation/stable/reference/generated/networkx.generators.line.line_graph.html)

It is possible that the graph contains two vertices, say, (1, 2) and (1, 3), and an edge connecting both: ((1, 2), (1, 3)). Thus, verifying that `v` is a tuple containing two elements is not enough to determine whether it represents an edge or a vertex.

One way I would solve this is by directly asking whether `v` is contained in the vertices or the edges before attempting to retrieve the Mobject:
```py
if v in self.vertices:
    return self.vertices[v]
if v in self.edges:
    return self.edges[v]
raise IndexError(f"Vertex or edge {v} not found")
# or some more elaborated message, like
# "Couldn't find vertex {v} or edge connecting vertices {v[0]} and {v[1]}"
# when v is a tuple of 2 elements
```

**Review by chopan050 — CHANGES_REQUESTED** [review]
Thanks for implementing this and adding tests! I left one comment which is necessary to address:

**Review by chopan050 — APPROVED** [review]
It turns out that an older PR, #3799, already fixed this issue with the solution I proposed.

However, we still need tests for this new feature, so your changes are still welcome. Thanks for this contribution!
